src/NLog/Targets/FileTarget.cs

```diff
@@ -1660,17 +1523,15 @@ private void ProcessOnStartup(string fileName, LogEventInfo logEvent)
                     InternalLogger.Warn("Unable to archive old log file '{0}': {1}", fileName, exception);
                 }
             }
+        }
 
-            if (this.DeleteOldFileOnStartup)
-            {
-                try
-                {
+        private void DeleteOnStartup(string fileName) {
+            if (this.DeleteOldFileOnStartup) {
```

Comment:
Fixed

I apologise for this. It was accidental. It have been caused my VS settings as I was moving the code around.